Sources_v3/Utils/Internet Connection/InternetConnectionTests.swift

```diff
@@ -0,0 +1,73 @@
+//
+// Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+@testable import StreamChatClient
+import XCTest
+
+class InternetConnectionTests: XCTestCase {
+    var monitor: InternetConnectionMonitorMock!
+    var delegate: InternetConnectionDelegateMock! // swiftlint:disable:this weak_delegate
+    var internetConnection: InternetConnection!
+    
+    override func setUp() {
+        super.setUp()
+        
+        monitor = InternetConnectionMonitorMock()
+        delegate = InternetConnectionDelegateMock()
+        internetConnection = InternetConnection(monitor: monitor)
+        internetConnection.delegate = delegate
+    }
+
+    func test_internetConnection_start() throws {
+        XCTAssertEqual(internetConnection.status, .unknown)
+        internetConnection.start()
+        XCTAssertTrue(monitor.isStarted)
+        XCTAssertEqual(delegate.statuses, [.available(.great)])
+        XCTAssertEqual(internetConnection.status, .available(.great))
+    }
+
+    func test_internetConnection_stop() throws {
+        internetConnection.start()
+        XCTAssertTrue(monitor.isStarted)
+        internetConnection.stop()
+        XCTAssertFalse(monitor.isStarted)
+        XCTAssertEqual(delegate.statuses, [.available(.great), .unknown])
+        XCTAssertEqual(internetConnection.status, .unknown)
+    }
```

Comment:
So, these tests are here to test the mock implementation below? If yes, they should be named `InternetConnectionMock_Tests`. Why do you think this is worth having, btw?

Reply:
I think it's good to have them, at least to be sure notifications works.